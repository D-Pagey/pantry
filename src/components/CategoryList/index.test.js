import React from 'react';
import { CategoryList } from '.';
import { Fridge } from '../../fixtures';

export const CategoriesWithCount = [
  {
      colour: 'red',
      id: '111',
      name: 'meat',
      count: 3
  },
  {
      colour: 'blue',
      id: '222',
      name: 'fish',
      count: 1,
  },
  {
      colour: 'pink',
      id: '444',
      name: 'snacks'
  }
];

const context = {
  categories: CategoriesWithCount,
  fridge: Fridge
};

describe('CategoryList component', () => {
  it('should render', () => {
    const { container } = render(<CategoryList />, context);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should only render categories with counts > 0', () => {
    const { getByTestId, queryByTestId } = render(<CategoryList />, context);
    getByTestId('categoryCardmeat');
    getByTestId('categoryCardfish');
    expect(queryByTestId('categoryCardsnacks')).toBeNull();
  });
});
