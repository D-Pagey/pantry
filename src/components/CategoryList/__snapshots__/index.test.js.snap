// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList component should render 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 3px solid orange;
  border-radius: 6px;
  box-sizing: border-box;
  box-shadow: rgba(79,90,109,0.3) 0px 0.1875rem 0.625rem 0px;
  color: initial;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1rem;
  width: 120px;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 3px solid blue;
  border-radius: 6px;
  box-sizing: border-box;
  box-shadow: rgba(79,90,109,0.3) 0px 0.1875rem 0.625rem 0px;
  color: initial;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1rem;
  width: 120px;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 3px solid red;
  border-radius: 6px;
  box-sizing: border-box;
  box-shadow: rgba(79,90,109,0.3) 0px 0.1875rem 0.625rem 0px;
  color: initial;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1rem;
  width: 120px;
}

.c2 {
  margin: 0;
  padding: 0 0 0.5rem;
}

.c3 {
  margin: 0;
}

.c0 {
  display: grid;
  grid-gap: 0.5rem;
  grid-template-columns: repeat(auto-fill,minmax(120px,1fr));
  justify-items: center;
  padding: 1rem 0;
}

<div
  class="c0"
  data-testid="categoryList"
>
  <a
    class="c1"
    colour="orange"
    data-testid="categoryCardall"
    href="/food/all"
  >
    <h3
      class="c2"
    >
      All
    </h3>
    <p
      class="c3"
    >
      4
    </p>
  </a>
  <a
    class="c4"
    colour="blue"
    data-testid="categoryCardfish"
    href="/food/fish"
  >
    <h3
      class="c2"
    >
      Fish
    </h3>
    <p
      class="c3"
    >
      1
    </p>
  </a>
  <a
    class="c5"
    colour="red"
    data-testid="categoryCardmeat"
    href="/food/meat"
  >
    <h3
      class="c2"
    >
      Meat
    </h3>
    <p
      class="c3"
    >
      3
    </p>
  </a>
</div>
`;
